sources/ClangSharp.PInvokeGenerator/PInvokeGenerator.cs

```diff
@@ -843,299 +729,336 @@ private string GetParmModifierForPointeeType(Decl decl, Type pointeeType)
-        {
-            switch (pointeeType.Kind)
-            {
-                case CXTypeKind.CXType_Void:
-                {
-                    if (_config.GenerateUnsafeCode)
+                    case "int16_t":
                     {
-                        return "void*";
+                        name = "short";
+                        break;
                     }
 
-                    _outputBuilder.AddUsingDirective("System");
-                    return "IntPtr";
-                }
+                    case "int32_t":
+                    {
+                        name = "int";
+                        break;
+                    }
 
-                case CXTypeKind.CXType_UChar:
-                case CXTypeKind.CXType_UShort:
-                case CXTypeKind.CXType_UInt:
-                case CXTypeKind.CXType_ULong:
-                case CXTypeKind.CXType_ULongLong:
-                case CXTypeKind.CXType_SChar:
-                case CXTypeKind.CXType_Short:
-                case CXTypeKind.CXType_Int:
-                case CXTypeKind.CXType_Long:
-                case CXTypeKind.CXType_LongLong:
-                case CXTypeKind.CXType_Float:
-                case CXTypeKind.CXType_Double:
-                case CXTypeKind.CXType_Pointer:
-                case CXTypeKind.CXType_Record:
-                case CXTypeKind.CXType_Enum:
-                case CXTypeKind.CXType_Typedef:
-                {
-                    switch (decl.Kind)
+                    case "int64_t":
+                    case "time_t":
                     {
-                        case CXCursorKind.CXCursor_FieldDecl:
-                        case CXCursorKind.CXCursor_FunctionDecl:
-                        {
-                            var name = "IntPtr";
-
-                            if (_config.GenerateUnsafeCode)
-                            {
-                                name = GetTypeName(decl, pointeeType);
-                                name += '*';
-                            }
-                            else
-                            {
-                                _outputBuilder.AddUsingDirective("System");
-                            }
-
-                            return name;
-                        }
+                        name = "long";
+                        break;
+                    }
 
-                        case CXCursorKind.CXCursor_ParmDecl:
-                        {
-                            var name = GetTypeName(decl, pointeeType);
+                    case "intptr_t":
+                    case "ptrdiff_t":
+                    {
+                        name = "IntPtr";
+                        break;
+                    }
 
-                            if (_config.GenerateUnsafeCode)
-                            {
-                                name += '*';
-                            }
-                            return name;
-                        }
+                    case "size_t":
```

Comment:
`size_t` is spec'd to be unsigned, so this is a change in how the bindings are generated.